roxctl/central/generate/generate.go

```diff
@@ -203,13 +202,22 @@ func createBundle(logger logger.Logger, config renderer.Config) (*zip.Wrapper, e
 		config.Environment[env.OfflineModeEnv.EnvVar()] = strconv.FormatBool(config.K8sConfig.OfflineMode)
 
 		if config.K8sConfig.Telemetry.Enabled {
-			logger.InfofLn(`Unless run in offline mode,
- StackRox Kubernetes Security Platform collects and transmits aggregated usage and system health information.
-  If you want to OPT OUT from this, re-generate the deployment bundle with the '--enable-telemetry=false' flag`)
+			if env.TelemetryStorageKey.Setting() == "" {
+				return errox.InvalidArgs.Newf("telemetry storage key is not set")
+			}
```

Comment:
Hm, I understand your reasoning of preparing this for when we actually enable the telemetry for on-prem installations, but right now this would be a breaking change for clients that still use the `--enable-telemetry=true` flag (previously, it did essentially nothing, and honestly it should have been either marked as deprecated or hidden, but now it will lead to a failure during generation).

I don't really expect many people to use `--enable-telemetry=true` to being with, but it might be better for now to be more lax here and not error out - keeping at the very least the same behavior as before: `--enable-telemetry=true` doesn't do anything _yet and has no side effects_.

Alternatively, in conjunction with the change of behavior above, we could also mark the flag as hidden right now: since it doesn't really provide any value yet, we could use `cobra.MarkFlagHidden` and `flags.NoInteractiveKey` to prevent the flag from being shown either within the `Usage` output or within the `generate interactive` prompts.

WDYT?

Reply:
We had to deprecate this flag much earlier. With this PR **it works again** and can be used (it's not only a preparation work): if the key is set in the environment, it will enable telemetry.

I could suggest issuying a warning instead, saying the telemetry won't be enabled due to missing storage key. Then, we'll switch to an error in the release when we enable telemetry for on-prem.